### PR TITLE
api created for number of voters

### PR DESCRIPTION
**Pull request by Suyog007**
The major tasks are:

- Created separate vardb for counting the voters who are against as well as with the proposal.
- Created an external function to return those values.

### CODE REVIEW

File: core_contracts/governance/governance.py [hunk 1]
```diff
@@ -138,15 +144,19 @@ def castVote(self, name: str, vote: bool) -> None:
         total_for_votes = proposal.total_for_votes.get()
         total_against_votes = proposal.total_against_votes.get()
         if vote:
+            total_for_voters_count = proposal.for_voters_count.get()
             proposal.for_votes_of_user[sender] = stake
             proposal.against_votes_of_user[sender] = 0
             total_for = total_for_votes + stake - prior_vote[0]
             total_against = total_against_votes - prior_vote[1]
+            proposal.for_voters_count.set(total_for_voters_count + 1)
```
Comment (dbrehmer) [inline core_contracts/governance/governance.py:152]: We need to check if the person has previously voted. If they are making the same vote we just return. If they are changing their vote we remove one from one total and add one to the other.

File: core_contracts/governance/governance.py [hunk 2]
```diff
@@ -53,6 +53,12 @@ def name(self) -> str:
     def getDay(self) -> int:
         return (self.now() - self._time_offset.get()) // U_SECONDS_DAY
 
+    @external(readonly=True)
+    def getVotersCount(self, name: str) -> dict:
```
Comment (dbrehmer) [inline core_contracts/governance/governance.py:57]: Let's also add these data to what is returned by checkVote().

**Review by dbrehmer — CHANGES_REQUESTED** [review]
Scott also wants a getProposals method that returns a batch of vote proposals. Let's include that in this PR.
It will take parameters _batch_size and _offset and return the data for _batch_size number of proposals. Maybe we also have a default to return the last 20.

File: core_contracts/governance/governance.py (hunk 2 above)
Comment (Suyog007) [reply core_contracts/governance/governance.py:57]: added!!

File: core_contracts/governance/governance.py [hunk 3]
```diff
@@ -134,19 +150,35 @@ def castVote(self, name: str, vote: bool) -> None:
         snapshot = proposal.vote_snapshot.get()
         baln = self.create_interface_score(self.addresses['baln'], BalancedInterface)
         stake = baln.stakedBalanceOfAt(sender, snapshot)
+        if stake == 0:
+            revert(f'Balanced tokens needs to be staked to cast the vote.')
```
Comment (dbrehmer) [inline core_contracts/governance/governance.py:154]: Will need to base this on the total voting power including LP'd BALN once merged with the branch that includes LP'd BALN.

**Review by dbrehmer — APPROVED** [review]
Will still need a change to revert on total voting power == zero rather than stake == 0.

**Review by dbrehmer — APPROVED** [review]
Looks fine.

**Pull request merged**